src/Microsoft.Extensions.Caching.Abstractions/IMemoryCache.cs

```diff
@@ -30,5 +30,11 @@ public interface IMemoryCache : IDisposable
         /// </summary>
         /// <param name="key">An object identifying the entry.</param>
         void Remove(object key);
+
+        /// <summary>
+        /// Attempt to remove the given percentage of the total entries in the cache.
+        /// </summary>
+        /// <param name="percentage">The percentage of cache entries to remove.</param>
+        void Compact(double percentage);
```

Comment:
Shifting to expose Count and using Compact to use specific numbers might be a good change.

@sebastienros we still have policies it's just not configurable or extensible right now. The Compact method removes based on 1) whether the entry is expired 2) CacheItemPriority 3) LRU. There was discussion on incorporating memory footprint into the triggering mechanism but the problem is that there's currently no accurate way of determining memory usage. As such it was determined to expose Compact on the interface and allow users or external triggers to compact the memory cache directly.